src/main/java/com/k3project/demo/controller/UserController.java

```diff
@@ -11,23 +12,49 @@
 import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.*;
 
+
 import java.util.*;
 
+@Tag(name = "User Management", description = "APIs for managing users")
 @RestController
 //define that all methods will returned Body in serialized Json format
-@RequestMapping(path = "/api/v1", produces = MediaType.APPLICATION_JSON_VALUE)
+@RequestMapping(path = "/api/v1/user", produces = MediaType.APPLICATION_JSON_VALUE)
 public class UserController {
     private final UserService userService;
 
-    public UserController( UserService userService) {
+    public UserController(UserService userService) {
         this.userService = userService;
     }
 
-    @GetMapping(value = "/allUsers")
+    /**
+     * Retrieves all users from the database.
+     * <p>
+     * {@code GET/users} : get all users from database
+     * This method delegates to {@link UserService#findAllUsers()} to fetch the users.
+     *
+     * @return A list of {@code UserDTO} objects representing all users.
+     * or with status {@code 400(Bad Request)} if the request is not valid,
+     * or with status {@code 404(Not Found)} if list of users is not found in database,
+     * or with status {@code 500(Internal Server Error)}when the server error was occurred,
+     * @see UserService#findAllUsers()
+     */
+    @GetMapping(value = "users")
     public List<UserDTO> findAllUsers() {
         return userService.findAllUsers();
     }
 
+    /**
+     * Retrieves a user by their email address.
+     * <p>
+     * {@code GET /email}: get User from database regarding to his email address
+     *
+     * @param email The email address of the user to retrieve.
+     * @return the {@link ResponseEntity} with status {@code 200(OK)} and the UserDTO is returned in Json body.
+     * or with status {@code 400(Bad Request)} if the request is not valid,
+     * or with status {@code 404(Not Found)} if email does not match with any email in database,
+     * or with status {@code 500(Internal Server Error)}when the server error was occurred
+     * @see UserService#findUserByEmail(String)
+     */
     @GetMapping("/email")
```

Comment:
@GetMapping("/user") ... @RequestParam is email
@GetMapping("/user/{email}")... @PathVariable String email 
Its up to you what you want to use, but use one style for every endpoint